project-base/src/Shopsys/ShopBundle/Resources/scripts/frontend/components/ajaxMoreLoader.js

```diff
@@ -35,21 +38,25 @@
 
         var onClickLoadMoreButton = function () {
             $(this).hide();
+
+            var requestData = {};
+            requestData[pageQueryParameter] = page + 1;
+
             Shopsys.ajax({
-                loaderElement: '.js-product-list-with-paginator',
+                loaderElement: $wrapper,
                 type: 'GET',
-                url: document.location,
-                data: {page: page + 1},
+                url: url,
+                data: requestData,
                 success: function (data) {
                     var $response = $($.parseHTML(data));
-                    var $nextProducts = $response.find('.js-product-list > li');
-                    $currentProductList.append($nextProducts);
+                    var $nextItems = $response.find('.js-list > li');
```

Comment:
I would remove the specific HTML tag if we're generalizing the component.
```suggestion
                    var $nextItems = $response.find('.js-list > *');
```